### PR TITLE
Connect screens to backend api

### PR DESCRIPTION
**Pull request by junchen-liu**
1. After manually testing, the Login Page can be connected to the backend Locally.
2. Fix some bugs in SignUpPage.
3. Set Login Page to be the initial Page,

### CODE REVIEW

**Comment by junchen-liu** [comment]
`pubspec.lock` is updated automatically, should I cancel this update?

File: lib/screens/displayphotos_screen.dart [hunk 1]
```diff
@@ -100,6 +102,55 @@ class _DisplayPhotoScreenState extends State<DisplayPhotoScreen> {
...
+      } catch (e) {
+        print(e);
+      }
+    }
+
+    //Uploading function
+    void _uploadImage() async {
+      print('continue');
+
+      final String address = "http://127.0.0.1/api";
```
Comment (IanSaucy) [inline lib/screens/displayphotos_screen.dart]: Dont statically code api urls. This should be set from some central location so that we can change it.

Comment (junchen-liu) [reply lib/screens/displayphotos_screen.dart]: Done. And I update the HttpService. Thanks for the suggestion!

File: lib/services/service.dart [hunk 2]
```diff
@@ -8,7 +8,7 @@ class ReturnHttpPost<T> {
 }
 
 class HttpService {
-  final String address = "http://127.0.0.1/api";
+  static final String address = "http://192.168.1.104/api";
```
Comment (junchen-liu) [inline lib/services/service.dart:11]: There will be [SocketException](https://stackoverflow.com/questions/55785581/socketexception-os-error-connection-refused-errno-111-in-flutter-using-djan) when using `127.0.0.1`, so I use my local ipv4 address.

File: lib/screens/displayphotos_screen.dart [hunk 3]
```diff
@@ -100,6 +103,55 @@ class _DisplayPhotoScreenState extends State<DisplayPhotoScreen> {
       );
     }
 
+    postData(filepaths, apiUrl) async {
+      try {
+        var request = http.MultipartRequest('POST', Uri.parse(apiUrl));
+
+        //TODO: request fields
+        request.fields['userid'] = '1';
+        request.fields['cooking_date'] = '2022-10-22';
```
Comment (IanSaucy) [inline lib/screens/displayphotos_screen.dart]: Is this still needed?

File: lib/screens/displayphotos_screen.dart [hunk 4]
```diff
@@ -100,6 +103,55 @@ class _DisplayPhotoScreenState extends State<DisplayPhotoScreen> {
...
+          print('failed');
+        }
+      } catch (e) {
+        print(e);
+      }
+    }
+
+    //Uploading function
+    void _uploadImage() async {
+      print('continue');
```
Comment (IanSaucy) [inline lib/screens/displayphotos_screen.dart]: Remove from final code

File: lib/screens/displayphotos_screen.dart [hunk 5]
```diff
@@ -100,6 +103,55 @@ class _DisplayPhotoScreenState extends State<DisplayPhotoScreen> {
...
+        print(res.toString());
+
+        res.stream.listen((value) {
+          print('value');
+          print(value);
+        });
+        if (res.statusCode == 200) {
+          print('uploaded');
+        } else {
+          print('failed');
```
Comment (IanSaucy) [inline lib/screens/displayphotos_screen.dart:133]: Remove from final code

File: lib/screens/signup_screen.dart [hunk 6]
```diff
@@ -18,7 +18,7 @@ final signupRepositoryProvider =
     Provider<SignupRepository>((ref) => SignupRepository());
 
 class SignupRepository {
-  var url = Uri.parse("http://192.168.0.44:8888/api/signup");
+  var url = Uri.parse("http://192.168.1.104/api/signup");
```
Comment (IanSaucy) [inline lib/screens/signup_screen.dart:24]: The base URL should come from a central setting. Be that an env variable, static const etc. That way you can change the URL of all queries at once.

**Review by IanSaucy — CHANGES_REQUESTED** [review]
Just a few small changes

File: lib/screens/displayphotos_screen.dart (hunk 3 above)
Comment (junchen-liu) [reply lib/screens/displayphotos_screen.dart]: No, I just deleted these. Thanks

File: lib/screens/signup_screen.dart (hunk 6 above)
Comment (junchen-liu) [reply lib/screens/signup_screen.dart:24]: Thanks for the suggestion. I agree with you. Also, The Sign Up Screen doesn't use the functions in `UserService` and `HttpService`. I think there will be another updating the signup screen including the URL part.

**Pull request merged**